core-libs/setup/ssr/optimized-engine/rendering-cache.spec.ts

```diff
@@ -151,4 +160,46 @@ describe('RenderingCache with cacheSize', () => {
       expect(renderingCache.get('c')).toBeTruthy();
     });
   });
+
+  describe('RenderingCache and cacheStrategyResolver', () => {
+    let renderingCache: RenderingCache;
+
+    describe('if default cacheStrategyResolver', () => {
+      it('should cache if avoidCachingErrors is false', () => {
+        renderingCache = new RenderingCache({
+          ...options,
+          avoidCachingErrors: false,
+        });
+        renderingCache.store('a', new Error('err'), 'a');
+        expect(renderingCache.get('a')).toMatchSnapshot();
+      });
+
+      it('should not cache if avoidCachingErrors is true', () => {
+        renderingCache = new RenderingCache({
+          ...options,
+          avoidCachingErrors: true,
+        });
+        renderingCache.store('a', new Error('err'), 'a');
+        expect(renderingCache.get('a')).toBeFalsy();
+      });
+    });
```

Comment:
1. To cover all cases, let's create tests for positive HTML results. As a result I expect the  number of tests to double.

2. let's mention `errors` in the test descriptions
```suggestion
    describe('if default cacheStrategyResolver', () => {
      it('should cache errors if avoidCachingErrors is false', () => {
        renderingCache = new RenderingCache({
          ...options,
          avoidCachingErrors: false,
        });
        renderingCache.store('a', new Error('err'), 'a');
        expect(renderingCache.get('a')).toMatchSnapshot();
      });

      it('should not cache errors if avoidCachingErrors is true', () => {
        renderingCache = new RenderingCache({
          ...options,
          avoidCachingErrors: true,
        });
        renderingCache.store('a', new Error('err'), 'a');
        expect(renderingCache.get('a')).toBeFalsy();
      });
    });
```